rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -165,6 +166,8 @@ def __init__(
         self.decoders: dict[str, 'DecoderInterface'] = {}
         # store the mapping of possible counterparties to the allowed types and subtypes in events
         self.events_types_tuples: DecoderEventMappingType = {}
+        # store the mapping of possible counterparties to their evm products
+        self.possible_products: dict[str, list[EvmProduct]] = {}
```

Comment:
Why store those?

I would say to just calculate them on the fly. The api will be called only once and then they will be stored in the frontend. So keeping them forever in memory here makes no sense.